solidity/contracts/ECDSARewardsDistributor.sol

```diff
@@ -86,9 +90,13 @@ contract ECDSARewardsDistributor is Ownable {
 
         // Mark it claimed and send the token.
         _setClaimed(merkleRoot, index);
-        require(IERC20(token).transfer(account, amount), "Transfer failed");
 
-        emit RewardsClaimed(merkleRoot, index, account, amount);
+        address beneficiary = tokenStaking.beneficiaryOf(account);
+        require(beneficiary != address(0), "Beneficiary address not set");
+
+        require(IERC20(token).transfer(beneficiary, amount), "Transfer failed");
+
+        emit RewardsClaimed(merkleRoot, index, account, beneficiary, amount);
```

Comment:
Since we're adding 'beneficiary', maybe we should rename 'account' to 'operator' to make it clear what account we have in mind.

Reply:
Yeah, I also vote for renaming `account` to `operator`, also as a parameter of `claim` function.